lang/cs/Org.Apache.REEF.Network/Naming/NameCache.cs

```diff
@@ -38,19 +39,35 @@ internal sealed class NameCache
         private readonly double _expirationDuration;
 
         [Inject]
+        [Obsolete("TODO[JIRA REEF-1856] This constructor will be removed")]
         private NameCache(
             [Parameter(typeof(NameCacheConfiguration.CacheEntryExpiryTime))] double expirationDuration,
             [Parameter(typeof(NameCacheConfiguration.CacheMemoryLimit))] string memoryLimit,
             [Parameter(typeof(NameCacheConfiguration.PollingInterval))] string pollingInterval)
         {
-            var config = new NameValueCollection
+            Logger.Log(Level.Warning, "Received a parameter \"PollingInterval\" which will be interpreted as ExpirationScanFrequency.");
+            Logger.Log(Level.Warning, "Received a parameter \"CacheMemoryLimit\" which will be ignored.");
+
+            var cacheConfig = new MemoryCacheOptions
+            {
+                ExpirationScanFrequency = TimeSpan.Parse(pollingInterval)
```

Comment:
I do assume that the defaults for these will be reasonable. I was hoping to restrict the API to as few options as possible (so as to not target this specific implementation), but given that these `MemoryCacheOptions` are for the `IMemoryCache` interface, shall I go ahead and add them in?